assets/src/edit-story/app/story/actions/useSaveStory.js

```diff
@@ -75,70 +75,73 @@ function useSaveStory({ storyId, pages, story, updateStory }) {
     );
   }, []);
 
-  const saveStory = useCallback(() => {
-    setIsSaving(true);
-    const {
-      title,
-      status,
-      author,
-      date,
-      modified,
-      slug,
-      excerpt,
-      featuredMedia,
-      password,
-      publisherLogo,
-    } = story;
+  const saveStory = useCallback(
+    (props) => {
+      setIsSaving(true);
+      const {
+        title,
+        status,
+        author,
+        date,
+        modified,
+        slug,
+        excerpt,
+        featuredMedia,
+        password,
+        publisherLogo,
+      } = story;
 
-    const content = getStoryMarkup(story, pages, metadata);
-    saveStoryById({
-      storyId,
-      title,
-      status,
-      pages,
-      author,
-      slug,
-      date,
-      modified,
-      content,
-      excerpt,
-      featuredMedia,
-      password,
-      publisherLogo,
-    })
-      .then((post) => {
-        const {
-          status: newStatus,
-          slug: newSlug,
-          link,
-          featured_media_url: featuredMediaUrl,
-        } = post;
-
-        updateStory({
-          properties: {
+      const content = getStoryMarkup(story, pages, metadata);
+      saveStoryById({
+        storyId,
+        title,
+        status: props?.status ? props.status : status,
+        pages,
+        author,
+        slug,
+        date,
+        modified,
+        content,
+        excerpt,
+        featuredMedia,
+        password,
+        publisherLogo,
+      })
+        .then((post) => {
+          const {
```

Comment:
We actually have `objectPick` in utils, which could make this a lot cleaner.

```js
      const propsToSave = objectPick(story, [
        'title',
        'status',
        'author',
        'date',
        'modified',
        'slug',
        'excerpt',
        'featuredMedia',
        'password',
        'publisherLogo',
      ]);
      const content = getStoryMarkup(story, pages, metadata);
      saveStoryById({
        storyId,
        ...propsToSave,
        ...props,
      });
```

Reply:
Definitely. Thanks @barklund